src/modules/Nfs.rb

```diff
@@ -362,21 +288,97 @@ def Read
-    # and starts the services together.
-    # (No parameters because it is too short to abort)
-    # @return true on success
-    def WriteOnly
-      # Merge with non-nfs entries from fstab:
-      fstab = SCR.Read(path(".etc.fstab"))
-      # unescape deferred for optimization
-      fstab.delete_if { |entry| ["nfs", "nfs4"].include?(entry["vfstype"]) }
-      fstab = UnescapeSpaces(fstab)
+    # Read /etc/fstab if this is running as a standalone YaST module.
+    #
+    # If it runs embedded into the expert partitioner, libstorage takes care of
+    # /etc/fstab, so this function does nothing.
+    #
+    def read_etc_fstab
+      # Skip if not running standalone, but in the expert partitioner:
+      # Then libstorage takes care of /etc/fstab.
+      return if @skip_fstab
+
+      fstab = EtcFstab.new(@etc_fstab_name)
+      nfs_entries = fstab.select { |e| e.fs_type.start_with?("nfs") }
+      @nfs_entries = nfs_entries.map do |entry|
+        share = {}
+        share["spec"]    = entry.device
+        share["file"]    = entry.mount_point
+        share["vfstype"] = entry.fs_type
+        share["mntops"]  = entry.format_mount_opts
+        share
+      end
+    end
+
+    # Write /etc/fstab if this is running as a standalone YaST module.
+    #
+    def write_etc_fstab
+      return if @skip_fstab
+      backup_etc_fstab
+      fstab = EtcFstab.new(@etc_fstab_name)
+      merge_entries(fstab)
+      remove_unknown_shares(fstab)
+      fstab.fix_mount_order
+      fstab.write
+    end
+
+    # Merge or add all @nfs_entries to an EtcFstab object.
+    #
+    # @param fstab [EtcFstab]
+    #
+    def merge_entries(fstab)
+      @nfs_entries.each do |share|
+        device = share["spec"]
+        entry = fstab.find_device(device)
+        if entry
+          fill_from_nfs_entry(entry, share)
+        else
+          entry = fstab.create_entry
+          entry.device = device
+          fill_from_nfs_entry(entry, share)
+          fstab.add_entry(entry)
+        end
+      end
+    end
+
+    # Remove all NFS entries from fstab that don't have a counterpart in
+    # @nfs_entries
+    #
+    # @param fstab [EtcFstab]
+    #
+    def remove_unknown_shares(fstab)
+      shares = @nfs_entries.map { |share| share["spec"] }
+      fstab.delete_if do |entry|
+        entry.fs_type.start_with?("nfs") && !shares.include?(entry.device)
+      end
+    end
+
+    # Fill an EtcFstab entry from an @nfs_entry hash.
+    #
+    # @param entry [EtcFstab::Entry]
+    # @param share [Hash]
+    #
+    def fill_from_nfs_entry(entry, share)
+      entry.mount_point = share["file"] || "/unknown_nfs"
+      entry.fs_type     = share["fs_type"] || "nfs"
+      mount_opts        = share["mntops"] || ""
+      entry.parse_mount_opts(mount_opts)
+    end
+
+    # Back up /etc/fstab to /etc/fstab.YaST2.save
+    #
+    def backup_etc_fstab
+      SCR.Execute(
+        path(".target.bash"),
+        "/bin/cp $ORIG $BACKUP",
```

Comment:
I don't know; that was taken literally from the old code. I just moved it to a separate function for clarity:

https://github.com/yast/yast-nfs-client/blob/SLE-12-SP3/src/modules/Nfs.rb#L393

As for only in the installed system - that's tricky since it also needs to work in both test suites, and that stuff already backfired on me several times when I tried to simplify things.